docs/async/async-csharp-vb.rst

```diff
@@ -0,0 +1,199 @@
+
+	private readonly HttpClient _httpClient = new HttpClient();
+
+	[HttpGet]
+	public async Task<int> MicrosoftCount()
+	{
+	    // Suspends MicrosoftCount() to allow the caller (the web service) to accept another request,
+	    // rather than blocking on this one.
+	    var html = await _httpClient.DownloadStringAsync("microsoft.com");
+
+	    return Regex.Matches(html, "Microsoft").Count;
+	}
+
+Example (VB)
+------------
+
+These are the VB-equivalent code snippets from above.
+
+Client app snippet (Universal Windows App):
+
+.. code-block:: vb.net
+
+	Private Readonly Dim _httpClient As HttpClient = new HttpClient()
+
+	Private Async Sub SeeTheMicrosofts_Click(sender As Object, e As EventArgs)
+
+	    ' Capture the task handle here so we can await it later.
+	    Dim getMicrosoftDotComHtmlTask As Task(Of String) = _httpClient.GetStringAsync("microsoft.com")
+
+	    ' Any other work on the UI thread can be done here, such as enabling a Progress Bar.
+	    NetworkProgressBar.IsEnabled = true
+	    NetworkProgressBar.Visibility = Visibility.Visible
+
+	    ' The await operator suspends SeeTheMicrosofts_Clicked, returning control to its caller.
+	    ' This is what allows the app to be responsive and not hang on the UI thread.
+	    Dim html As String = Await getMicrosoftDotComHtmlTask
+	    Dim count As Integer = Regex.Matches(html, "Microsoft").Count
+
+	    MicrosoftCountLabel.Text = $"Number of Microsofts on microsoft.com: {count}"
+
+	    NetworkProgressBar.IsEnabled = false
+	    NetworkProgressBar.Visbility = Visibility.Collapsed
+	End Sub
+
+Web Service snippet (ASP.NET MVC):
+
+.. code-block:: vb.net
+
+	Private Readonly Dim _httpClient As HttpClient = new HttpClient()
+
+	<HttpGet>
+	Public Async Function MicrosoftCount() As Task(Of String)
+
+	    ' Suspends MicrosoftCount() to allow the caller (the web service) to accept another request,
+	    ' rather than blocking on this one.
+	    Dim html As String = Await _httpClient.GetStringAsync("microsoft.com");
+
+	    Return Regex.Matches(html, "Microsoft").Count
+	End Function
+
+More on Task and Task<T>
+------------------------
+
+As mentioned before, Tasks are constructs used to represent operations which you will need to wait on to complete.  Although Tasks are general, the important context here is using them to represent an I/O operation.
+
+* ``Task`` represents a single operation which does not return a value.
+* ``Task<T>`` represents a single operation which returns a value of type ``T``.
+
+It's important to reason about Tasks as abstractions of work to be done rather than threads, because in the context of asynchronous I/O they aren't threads at all!  Asynchronous .NET library calls essentially delegate work to the operating system, which is naturally asynchronous when I/O is performed.  Unless you write code using a Task to abstract the notion of a new thread (such as the ``Task.Run`` method), there will be no new thread spawned as a result of your async code.
```

Comment:
> because in the context of asynchronous I/O they aren't threads at all

They are not ever threads.

Try to avoid the word "essentially" when it doesn't really add anything. 

Lots of .NET APIs delegate work to the OS. In this case, the delegation is to a low-level file system driver that calls back when there is work to do. My understanding is that the file system has both sync and async modes. Assuming thats true, we should avoid the word "naturally".

The last sentence is fine. It might be improved by saying "Tasks are always executed (in terms of CPU) on the current thread, unless a separate thread is requested by using the Task.Run method."